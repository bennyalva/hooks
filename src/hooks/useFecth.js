import React, { useState, useEffect, useRef } from 'react'

export const useFecth = (url) => {

 const isMounted = useRef(true);
 const [state, setstate] = useState({data: null, loading: true, error: null});
 useEffect(() => {
   return () => {
     isMounted.current = false;
   }
 }, [])
 useEffect(() => {
    setstate({data: null, loading: true, error: null})
    fetch(url)
    .then(result => result.json())
    .then(data => {
      setTimeout(() => {
          if(isMounted.current){
            setstate({
              loading: false,
              error: null,
              data
          })
          } else {
            console.log('no se llamo')
          }
      }, 1000)
    }).catch(() => setstate({
      data: null,
      loading: false,
      error: 'any error'
    }));
 }, [url]);
 return state;
}
