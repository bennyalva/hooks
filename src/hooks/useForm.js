import React, { useState } from 'react'

export const useForm = (initial = {}) => {
    const [values, setValues] = useState(initial);

    const handlerInputChange = ({target}) => {
        setValues({
            ...values,
            [target.name]: target.value
        })
    }

    return [values, handlerInputChange]
}
