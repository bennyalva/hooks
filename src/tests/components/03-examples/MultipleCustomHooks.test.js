import React from 'react';
import { useFecth } from '../../../hooks/useFecth';
import { useCounter } from '../../../hooks/useCounter';
const { shallow } = require("enzyme");
const { MultipleCustomHooks } = require("../../../components/03-examples/MultipleCustomHooks");
jest.mock('../../../hooks/useFecth')
jest.mock('../../../hooks/useCounter')
describe('Tests MultipleCustomHooks', () => {
    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    })
   test('should to show correctly', () => {
       useFecth.mockReturnValue({
           data: null,
           loading: true,
           error: null
       });
       const wrapper = shallow(<MultipleCustomHooks />);
       expect(wrapper).toMatchSnapshot();
   })
   test('should to show Info', () => {
        useFecth.mockReturnValue({
            data: [
                {
                    author: 'lbenny',
                    quote: 'Heyy rey'
                }
            ],
            loading: false,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Heyy rey');
        expect(wrapper.find('footer').text().trim()).toBe('lbenny')
   });
   
    
});
