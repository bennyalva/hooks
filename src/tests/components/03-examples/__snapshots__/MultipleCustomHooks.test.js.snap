// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests MultipleCustomHooks should to show correctly 1`] = `
<div>
  <h1>
    Breaking bad
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
